server/src/main/java/org/elasticsearch/common/io/stream/RecyclerBytesStreamOutput.java

```diff
@@ -43,9 +43,7 @@ public class RecyclerBytesStreamOutput extends BytesStream implements Releasable
 
     public RecyclerBytesStreamOutput(Recycler<BytesRef> recycler) {
         this.recycler = recycler;
-        try (Recycler.V<BytesRef> obtain = recycler.obtain()) {
-            pageSize = obtain.v().length;
-        }
+        this.pageSize = recycler.pageSize();
```

Comment:
Not super needed to cache this into a field now, but since we use this thing both backed by the Netty and big arrays allocator I figured I'd not introduce a potentially slow callsite from using this in `position()`

Reply:
:+1: we `assert pageSize == newPage.v().length;` when acquiring actual pages, I'd be uncomfortable if we lost that check.